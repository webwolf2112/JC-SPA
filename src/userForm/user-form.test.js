import React from 'react';
import { shallow } from 'enzyme';
import UserForm from './user-form';

describe( 'User Form Test ', () => {
    const defaultProps = {
        handleSubmit: jest.fn(),
        currentUser: {},
        cancel: jest.fn(),
        isFetching: false,
    };

    test('Should render correct', () => {
        const wrapper = shallow(
          <UserForm {...defaultProps} />,
        );
      
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find( 'button[type="submit"]' ).props().disabled ).toBeFalsy();
      });

      test('Should render correctly when fetching', () => {
        const wrapper = shallow(
          <UserForm {...defaultProps} isFetching={true} />,
        );
    
        expect( wrapper.find( 'button[type="submit"]' ).props().disabled ).toBeTruthy();
      });

      /** Due to a busy schedule I do not have time to complete all of the tests, but these
       * are the additional test cases I would normally write. 

        1) ensure the inputs populate correctly for update user
        2) ensure that the firstname input does not show up on update user
        3) ensure handle submit does not get called when the form does not pass validation
      **/
});

