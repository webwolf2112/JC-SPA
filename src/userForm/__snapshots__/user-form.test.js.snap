// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Form Test  Should render correct 1`] = `
<div
  className="modalContainer"
>
  <div
    className="modalContent"
  >
    <button
      aria-label="Close Modal"
      className="close"
      onClick={[MockFunction]}
    >
      X
    </button>
    <form
      className="form"
      onChange={[Function]}
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        User Name
      </label>
      <input
        id="username"
        required={true}
        type="text"
      />
      <label
        htmlFor="firstname"
      >
        First Name
      </label>
      <input
        id="firstname"
        type="text"
      />
      <label
        htmlFor="lastname"
      >
        Last Name
      </label>
      <input
        id="lastname"
        type="text"
      />
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        required={true}
        type="email"
      />
      <div
        className="buttonContainer"
      >
        <button
          aria-label="Cancel"
          className="inverse"
          onClick={[MockFunction]}
        >
          Cancel
        </button>
        <button
          disabled={false}
          type="submit"
        >
          Create User
        </button>
      </div>
    </form>
  </div>
</div>
`;
